runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpClientFactory.java

```diff
@@ -956,13 +953,16 @@ private void flushNext()
             {
                 final MutableDirectBuffer headerBuffer = bufferPool.buffer(headerSlot);
                 final long streamId = headerBuffer.getLong(headerSlotOffset);
-                final int length = headerBuffer.getInt(headerSlotOffset + Long.BYTES);
-                HttpExchange httpExchange = exchanges.get(streamId);
+                final long traceId = headerBuffer.getLong(headerSlotOffset + Long.BYTES);
+                final long authorization = headerBuffer.getLong(headerSlotOffset + 2 * Long.BYTES);
+                final int length = headerBuffer.getInt(headerSlotOffset +  3 * Long.BYTES);
+                final int metadataOffset = headerSlotOffset + 3 * Long.BYTES + Integer.BYTES;
```

Comment:
Let's use a flyweight for this, similar to `KafkaCacheEntry`, instead of manually calculating the offsets.